src/core/prompts/system-prompt/variants/native-next-gen/template.ts

```diff
@@ -48,13 +48,19 @@ export const BASE = `{{${SystemPromptSection.AGENT_ROLE}}}
 
 {{${SystemPromptSection.USER_INSTRUCTIONS}}}`
 
-const RULES = (_context: SystemPromptContext) => `RULES
+const RULES = (context: SystemPromptContext) => `RULES
 
-- The current working directory is \`{{CWD}}\` - this is the directory where all the tools will be executed from.`
+- The current working directory is \`{{CWD}}\` - this is the directory where all the tools will be executed from.${
+	context.enableParallelToolCalling
+		? `
+- You may use multiple tools in a single response when the operations are independent (e.g., reading several files, creating independent files). For dependent operations where one result informs the next, use tools sequentially and wait for the user's response.`
+		: ""
+}{{BROWSER_WAIT_RULES}}
+- MCP operations should be used one at a time, similar to other tool usage. Wait for confirmation of success before proceeding with additional operations.`
```

Comment:
we should only include this when !!context.mcpHub is available?
```suggestion
{!!context.mcpHub ? "- MCP operations should be used one at a time, similar to other tool usage. Wait for confirmation of success before proceeding with additional operations." : ""}`
```